### PR TITLE
Proposal practices testing

### PR DESCRIPTION
**Pull request by curtisblackwell**
## Author checklist
- [x] I'm proposing a single change in this PR.
- I believe the proposed standard:
  - [x] benefits more than it costs us and
  - [x] doesn't conflict with our people-first value.
- [x] The standard's file(s) follow(s) [the established directory structure][directory-structure].
- I provided:
  - [x] a succinct summary,
  - [x] reasoning for the proposed change,
  - [x] an example/instructions on how to embody the standard unless repealing, and
  - [x] a longer explanation if appropriate.

*Check boxes if requirement met or irrelevant.*

## First reviewer checklist
***NOTE:** Some of the author's checklist items are intentionally omitted, because they are a matter of opinion and up for discussion with the whole team.*

- [ ] This PR contains a single change.
- [ ] The proposed standard follows [the established directory structure][directory-structure].
- The author provided:
  - [ ] a succinct summary,
  - [ ] reasoning for the proposed change, and
  - [ ] an example/instructions on how to embody the standard unless repealing.

*Check boxes if requirement met or irrelevant.*

[directory-structure]: ../tree/master/standards#directory-structure

### CODE REVIEW

**Comment by curtisblackwell** [comment]
@mholmes-hs I added some examples, added some content, and restructured things to try and set clearer, more specific expectations.

It might be good to discuss things together after you have a chance to review my suggestions.

I also have these notes for myself that I'd be curious to hear your thoughts on:

shortcomings
- backend-focused (not a huge problem right now, b/c I'm pretty sure we don't have any test-writing devs who don't work on the backend)
- long (though for something as important as testing, I think it's probably fine)

ways to improve
- add macros for popular editors to simplify test formatting
- add testing diamond
    - few acceptance tests at the top: slowest, most confidence
    - mostly integration tests in the middle: medium speed and confidence
    - few unit tests at the bottom: fastest, not much confidence

File: standards/practices/testing.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+5. The tests cover complex functions likely affected by regressions.
+
+
+## Explanation
+Tests help us improve the quality of our software. However, tests carry maintenance costs, so **our tests should add value to our software**. In pursuit of this value:
+
+1. **Avoid overly-specific coverage goals.**  
+  E.g. 80% code coverage. Goals like this incentivize writing more tests than required to obtain confidence in the software.
+2. **Don't test what you don't own.**  
+  Testing frameworks and other third-party software indicates you don't trust the package you decided to rely on.
```
Comment (mholmes-hs) [inline standards/practices/testing.md]: I feel like we can have a 3rd point here.  We shouldn't try to test overly simple code.  In my original write up I said "don't test 1+1=2".  You put something similar: "The code covered is so simple it's not worth testing (little more than third-party code hidden behind a simple ternary)."

File: standards/practices/testing.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+Breaking these guidelines usually adds to maintenance costs without adding any value to our software.
```
Comment (mholmes-hs) [inline standards/practices/testing.md]: instead of "any value" could we say "worthwhile value"?

File: standards/practices/testing.md (hunk 1 above)
Comment (curtisblackwell) [reply standards/practices/testing.md]: Yeah, I want to make that same point w/o actually saying "1 + 1 = 2" b/c it's easy for people to argue past each other with an example that (hopefully) abstract.

What do you think of this?

3. **Avoid testing basic operations.**  
  Testing that PHP itself functions properly wastes effort and dilutes test suite value.

File: standards/practices/testing.md (hunk 2 above)
Comment (curtisblackwell) [reply standards/practices/testing.md]: Yes.

**Pull request merged**